### PR TITLE
Improve domain check

### PR DESCRIPTION
**Pull request by howard-stearns**
- simply tester questions
- include crowd summoning, with control over number and number making noise

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: f7742b99a9709ec3a57b42ce83cb5c7367015421 available at: https://gist.github.com/659a3b1d3d5dddf9d9f8e2bd6aa4cf98<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8908-f7742b9.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8908-f7742b9.dmg

**Comment by howard-stearns** [comment]
To test:
- Run 24 instances of http://hifi-content.s3.amazonaws.com/howard/scripts/tests/performance/crowd-agent.js?v=3 on some domain.  Wait for them to all spin up (checking the "Nodes" tab in settings to see that there are no queued agents and that they're all in the active (non-queued) section of the page). Note: they're already running on dev-playa as of this writing.
- Run developer/tests/performance/domain-check.js from this pr, with various input choices at the prompts.

**There is currently an issue with sound, being worked on separately. For now, I recommend you choose "Cancel" button, or 0 or blank input if/when prompted for "Number making sound?"**

Otherwise, all inputs should work:
- After inputs are collected, it should load up the content from localhost and "welcome" (for your version, e.g. dev-welcome), and then time the loading of the domain you requested. (If you request to test localhost or dev-welcome, it will omit the preloading of the requested domain.)
- After loading the content, it will check render rate during two strafing rotations, while simultaneously:
 -- dropping 100 tribbles  in front of you if you asked for them, in which case they should bounce around randomly, interact with physical objects, and change color smoothly. (They will stop moving after 1 minute, and the disappear after another minute. So you only need temp rez permission in the test domain.)
 -- summoning a bunch of avatars surrounding your initial position. The number will be the number you request minus the number already present (e.g., typically you and the sound searcher). The avatars should be in idle animation (although it may take a while before they all download the animation).

File: scripts/developer/tests/performance/domain-check.js [hunk 1]
```diff
@@ -133,48 +252,59 @@ function doRender(continuation) {
...
+                  HMD.active ? EXPECTED_HMD_FRAMERATE : EXPECTED_DESKTOP_FRAMERATE,
+                  HMD.active ? MINIMUM_HMD_FRAMERATE : MINIMUM_DESKTOP_FRAMERATE);
+        var total = AvatarList.getAvatarIdentifiers().length;
+        if (MINIMUM_AVATARS && !fail) {
+            if (0 === summonedAgents.length) {
+                fail = "FAIL: No agents reported.\n\Please run " + MINIMUM_AVATARS + " instances of\n\
+http://hifi-content.s3.amazonaws.com/howard/scripts/tests/performance/crowd-agent.js?v=3\n\
+on your domain server.";
+            } else if (total < MINIMUM_AVATARS) {
+                fail = "FAIL: Only " + summonedAgents.length + " of the expected " + (MINIMUM_AVATARS - total) + " agents reported in.";
```
Comment (davidkelly) [inline scripts/developer/tests/performance/domain-check.js]: the expected number of agents is 25 (by default) minus number of avatars present before you ran the script.  I think the above `MINIMUM_AVATARS - total` gives the number missing, not the number expected.

super minor but worth fixing

**Comment by davidkelly** [comment]
First time I ran this, I have 24 instances of crowd-agent.js running, but at the end it reported a failure with 23 reporting.  Also the error message said 23 of 1 -- probably need to reword that.  Oddly enough, 23 added avatars passed the next several times (using 25 as the total number to be present).  Unsure why once it said there was a fail.  I'd call this a QA pass nonetheless since I cannot reproduce it so far.  But the wording was definitely not right, and simple to fix.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8c5cdd910994363f50c9fe1855d1e58cbfba365 available at: https://gist.github.com/659a3b1d3d5dddf9d9f8e2bd6aa4cf98<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8908-d8c5cdd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8908-d8c5cdd.dmg

**Comment by howard-stearns** [comment]
thanks. updated the error message and ran through jslint.
The test failure may have been related to some problems I've seen with the default-home-content ac script for sound. It seems to crash and restart a lot, but I haven't looked into it yet. Maybe the numbers got off if that other ac was restarting at the wrong time?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 58d14decbbeea5176f90217b49cc118c03dc7e26 available at: https://gist.github.com/659a3b1d3d5dddf9d9f8e2bd6aa4cf98<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8908-58d14de.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8908-58d14de.dmg

**Pull request merged**